plugins/gitlab/api.mli

```diff
@@ -0,0 +1,75 @@
+module Status : sig
+  type t
+  type state = [ `Cancelled | `Failure | `Pending | `Running | `Success ]
+  val v : name:string -> ?description:string -> ?url:Uri.t -> state -> t
+end
+
+module Commit : sig
+  type t
+  val id : t -> Current_git.Commit_id.t
+  val repo_id : t -> Repo_id.t
+  val owner_name : t -> string
+  val hash : t -> string
+  val committed_date : t -> string
+  val pp : t Fmt.t
+  val compare : t -> t -> int
+  val set_status : t Current.t -> string -> Status.t Current.t -> unit Current.t
+  val uri : t -> Uri.t
+end
+
+module Ref : sig
+  type t = [ `PR of int | `Ref of string ]
+  val compare : t -> t -> int
+  val pp : t Fmt.t
+  val to_git : t -> string
+end
+
+module Ref_map : Map.S with type key = Ref.t
+
+type t
+type refs
+
+val of_oauth : token:string -> webhook_secret:string -> t
+val head_commit : t -> Repo_id.t -> Commit.t Current.t
+val refs : t -> Repo_id.t -> refs Current.Primitive.t
+val default_ref : refs -> string
+val webhook_secret : t -> string
+val all_refs : refs -> Commit.t Ref_map.t
+val ci_refs : ?staleness:Duration.t -> t -> Repo_id.t -> Commit.t list Current.t
+val cmdliner : t Cmdliner.Term.t
```

Comment:
Like in the `current_github` plugin, it would be nice if it was a `t option Cmdliner.Term.t`, so that a fallback could be set up in the absence of API keys.

Reply:
Agree, I'll make the change. The integration model with GitLab is quite different from GitHub.